frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/EventListenerDetails.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { SectionHeading, ResourceSummary } from '@console/internal/components/utils';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { TriggerTemplateModel, TriggerBindingModel } from '../../../models';
+import ResourceLinkList from '../resource-overview/ResourceLinkList';
+
+export interface EventListenerDetailsProps {
+  obj: K8sResourceKind;
+}
+
+const EventListenerDetails: React.FC<EventListenerDetailsProps> = ({ obj: eventListener }) => {
+  const resources = { template: [], bindings: [] };
+  const { template, bindings } =
+    eventListener?.spec?.triggers?.reduce(
+      (res, trigger) => {
+        res.template.push(trigger.template.name);
+        trigger.bindings.map((b) => res.bindings.push(b.name));
+        return res;
+      },
+      { ...resources },
+    ) || resources;
```

Comment:
```suggestion
const { template, bindings } =
  (eventListener.spec.triggers || []).reduce(
    (acc, trigger) => ({
      bindings: [...acc.bindings, trigger.bindings],
      template: [...acc.template, trigger.template],
    }),
    { template: [], bindings: [] },
  );
```

Rule of thumb is not to mutate outside resources during a `.map` or `.reduce` as they are intended to be as pure as possible. It's just easier for others to rationalize them out if you keep to these guidelines.

Also, `EventListenerKind` has `spec.triggers` as being a required path. So we either have an invalid type or unnecessary null-checks. 

One last thing is that `.reduce` will return the default if the array is empty so this structure allows you not to need to do an extra `||`.

Reply:
updated it.